Patch-Source: https://gitlab.com/ubports/porting/community-ports/android12/generic/kernel-android-common/-/commits/common-android12-5.10-halium (excluding apparmor changes)
---
From 70f103731d11d72a957fe94fd23a9c77711e4f8f Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Sat, 15 Oct 2022 11:20:36 +0000
Subject: [PATCH] (halium) GKI: use Android ABI padding for SYSVIPC task_struct
 fields

This allows to enable CONFIG_SYSVIPC=y without breaking module ABI
---
 include/linux/sched.h | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index d3cc279a4639..a52644d56bac 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -969,8 +969,10 @@ struct task_struct {
 	struct nameidata		*nameidata;
 
 #ifdef CONFIG_SYSVIPC
-	struct sysv_sem			sysvsem;
-	struct sysv_shm			sysvshm;
+	// struct sysv_sem			sysvsem;
+	/* sysvsem is in the ANDROID_KABI_RESERVE(1) field below */
+	// struct sysv_shm			sysvshm;
+	/* sysvshm is in the ANDROID_KABI_RESERVE(1) field below */
 #endif
 #ifdef CONFIG_DETECT_HUNG_TASK
 	unsigned long			last_switch_count;
@@ -1383,9 +1385,18 @@ struct task_struct {
 	ANDROID_KABI_RESERVE(3);
 	ANDROID_KABI_RESERVE(4);
 	ANDROID_KABI_RESERVE(5);
+
+#if defined(CONFIG_SYSVIPC)
+	// struct sysv_sem			sysvsem;
+	ANDROID_KABI_USE(6, struct sysv_sem sysvsem);
+	// struct sysv_shm			sysvshm;
+	_ANDROID_KABI_REPLACE(ANDROID_KABI_RESERVE(7); ANDROID_KABI_RESERVE(8),
+						  struct sysv_shm sysvshm);
+#else
 	ANDROID_KABI_RESERVE(6);
 	ANDROID_KABI_RESERVE(7);
 	ANDROID_KABI_RESERVE(8);
+#endif
 
 	/*
 	 * New fields for task_struct should be added above here, so that
-- 
GitLab

From b091c58ba8acbd51ba67aae28c2685c050692296 Mon Sep 17 00:00:00 2001
From: NeKit <nekit1000@gmail.com>
Date: Mon, 30 Mar 2020 02:22:41 +0200
Subject: [PATCH] (halium) binder: ignore txn_security_ctx flag

Needed for unpatched vndservicemanager to work with libselinux_stubs,
as it still sets FLAT_BINDER_FLAG_TXN_SECURITY_CTX.
---
 drivers/android/binder.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/android/binder.c b/drivers/android/binder.c
index 75072d3bc98c..01178fe716e5 100644
--- a/drivers/android/binder.c
+++ b/drivers/android/binder.c
@@ -882,7 +882,7 @@ static struct binder_node *binder_init_node_ilocked(
 	node->min_priority = to_kernel_prio(node->sched_policy, priority);
 	node->accept_fds = !!(flags & FLAT_BINDER_FLAG_ACCEPTS_FDS);
 	node->inherit_rt = !!(flags & FLAT_BINDER_FLAG_INHERIT_RT);
-	node->txn_security_ctx = !!(flags & FLAT_BINDER_FLAG_TXN_SECURITY_CTX);
+	node->txn_security_ctx = 0; /*!!(flags & FLAT_BINDER_FLAG_TXN_SECURITY_CTX);*/
 	spin_lock_init(&node->lock);
 	INIT_LIST_HEAD(&node->work.entry);
 	INIT_LIST_HEAD(&node->async_todo);
-- 
GitLab

From 2d859c22e9534241dbcdfa5a92f1cbb0cef163f9 Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Thu, 24 Mar 2022 14:58:20 +0200
Subject: [PATCH] bluetooth: hci: don't send READ_SYNC_TRAIN_PARAMS command

BT chip returns error for HCI_OP_READ_SYNC_TRAIN_PARAMS,
which causes the initialization to fail. As this feature is not used
anywhere, skipping reading the parameters should be safe.

Inspired by https://github.com/gemian/gemini-linux-kernel-3.18/commit/5a77c4336a50f446c97db6a286cf898422c1dcec
---
 net/bluetooth/hci_core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/net/bluetooth/hci_core.c b/net/bluetooth/hci_core.c
index a7fcfa30ab0b..df3f00b975bb 100644
--- a/net/bluetooth/hci_core.c
+++ b/net/bluetooth/hci_core.c
@@ -843,9 +843,14 @@ static int hci_init4_req(struct hci_request *req, unsigned long opt)
 	if (hdev->commands[30] & 0x08)
 		hci_req_add(req, HCI_OP_GET_MWS_TRANSPORT_CONFIG, 0, NULL);
 
+	/* Halium: BT chip returns error for HCI_OP_READ_SYNC_TRAIN_PARAMS,
+	 * which causes the initialization to fail. As this feature is not used
+	 * anywhere, skipping reading the parameters should be safe. */
+#if 0
 	/* Check for Synchronization Train support */
 	if (lmp_sync_train_capable(hdev))
 		hci_req_add(req, HCI_OP_READ_SYNC_TRAIN_PARAMS, 0, NULL);
+#endif
 
 	/* Enable Secure Connections if supported and configured */
 	if (hci_dev_test_flag(hdev, HCI_SSP_ENABLED) &&
-- 
GitLab
